src/main/java/org/openlmis/stockmanagement/dto/referencedata/WrappedOrderablesDto.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
+
+package org.openlmis.stockmanagement.dto.referencedata;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.openlmis.stockmanagement.web.Pagination;
+import org.springframework.data.domain.Page;
+
+public class WrappedOrderablesDto {
+
+  private List<OrderableDto> orderables;
+  private List<UUID> identifiers;
+
+  public WrappedOrderablesDto() {
+    this.orderables = new ArrayList<>();
+    this.identifiers = new ArrayList<>();
+  }
+
+  public void addEntry(ApprovedProductDto entry) {
```

Comment:
make this method private and require a collection of approved product in the constructor. Take a look at [ApproveProductsAggregator](https://github.com/OpenLMIS/openlmis-requisition/blob/master/src/main/java/org/openlmis/requisition/service/referencedata/ApproveProductsAggregator.java) in the [requisition](https://github.com/OpenLMIS/openlmis-requisition) service.